### PR TITLE
Fixes #31958: Support Ruby 2.7

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

**Comment by jlsherrill** [comment]
i had opened this if you want to Fixes it

**Comment by jlsherrill** [comment]
[test katello]

**Comment by ehelms** [comment]
@jlsherrill @adamruzicka I need a little help here. On Ruby 2.7 tests, there is only a single test failure, and its not clear to me (after trying a few things) where the fix needs to be: katello or foreman-tasks

```
ForemanTasks::TaskError: Task 6d539bb1-c958-4786-989b-970896bc6f64: ArgumentError: invalid byte sequence in UTF-8...
Stacktrace
ForemanTasks::TaskError: Task 6d539bb1-c958-4786-989b-970896bc6f64: ArgumentError: invalid byte sequence in UTF-8
    foreman-tasks (4.0.0) lib/foreman_tasks.rb:59:in `block in sync_task'
    foreman-tasks (4.0.0) lib/foreman_tasks.rb:58:in `tap'
    foreman-tasks (4.0.0) lib/foreman_tasks.rb:58:in `sync_task'
    /home/jenkins/workspace/katello-pr-test-ehelms/test/actions/pulp/repository/test_base.rb:15:in `run_action'
    /home/jenkins/workspace/katello-pr-test-ehelms/test/actions/pulp/repository/upload_file_test.rb:15:in `block in test_upload_file'
    /home/jenkins/workspace/katello-pr-test-ehelms/test/actions/pulp/repository/upload_file_test.rb:14:in `test_upload_file'
    /home/jenkins/workspace/katello-pr-test-ehelms/test/support/vcr.rb:26:in `block in run'
    /home/jenkins/workspace/katello-pr-test-ehelms/test/support/vcr.rb:25:in `run' (ForemanTasks::TaskError)
```

**Comment by jlsherrill** [comment]
here's the actual traceback from test.log:

```
2021-03-07T13:03:33 [E|bac|] invalid byte sequence in UTF-8 (ArgumentError)
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:208:in `gsub'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:208:in `filter_sensitive_data'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:204:in `log_message'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:216:in `log_exception'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:85:in `rescue in call'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/base.rb:34:in `call'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/runcible-2.13.1/lib/runcible/resources/content.rb:30:in `upload_bits'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/pulp/repository/upload_file.rb:14:in `block in run'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/pulp/repository/upload_file.rb:11:in `open'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/pulp/repository/upload_file.rb:11:in `run'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/action.rb:571:in `block (3 levels) in execute_run'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware/stack.rb:27:in `pass'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware.rb:19:in `pass'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware.rb:32:in `run'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware/stack.rb:23:in `call'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware/stack.rb:27:in `pass'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware.rb:19:in `pass'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/middleware/remote_action.rb:16:in `block in run'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/middleware/remote_action.rb:40:in `block in as_remote_user'
 | /home/jenkins/workspace/katello-pr-test/app/models/katello/concerns/user_extensions.rb:21:in `cp_config'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/middleware/remote_action.rb:27:in `as_cp_user'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/middleware/remote_action.rb:39:in `as_remote_user'
 | /home/jenkins/workspace/katello-pr-test/app/lib/actions/middleware/remote_action.rb:16:in `run'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware/stack.rb:23:in `call'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware/stack.rb:27:in `pass'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/dynflow-1.4.7/lib/dynflow/middleware.rb:19:in `pass'
 | /usr/local/rvm/gems/ruby-2.5.1@katello-pr-test-6318-3/gems/foreman-tasks-4.0.0/app/lib/actions/middleware/rails_executor_wrap.rb:14:in `block in run'
```

unclear why thats happening, it seems like some other exception is thrown (unexpectedly) and logging of it throws this error.  

We should probably investigate it as we'll be supporting pulp2 capsule syncing for some time (at least figure out what the exception is), but in the end we can probably just delete this test and related code since we don't support pulp2 uploading files anymore.

**Comment by ehelms** [comment]
For this, I opted to delete the particular test. I'd want to get this change merged and then make the change to CI structure via https://github.com/theforeman/jenkins-jobs/pull/16

File: katello.gemspec [hunk 1]
```diff
@@ -13,7 +13,7 @@ Gem::Specification.new do |gem|
   gem.homepage    = "http://www.katello.org"
   gem.summary     = "Content and Subscription Management plugin for Foreman"
   gem.description = "Katello adds Content and Subscription Management to Foreman. For this it relies on Candlepin and Pulp."
-  gem.required_ruby_version = ['>= 2.5.0', '< 2.7.0']
+  gem.required_ruby_version = ['>= 2.5.0', '< 2.8.0']
```
Comment (ekohl) [inline katello.gemspec:16]: Perhaps this is equivalent?
```suggestion
  gem.required_ruby_version = '~> 2.5'
```

**Review by jlsherrill — APPROVED** [review]

**Comment by ehelms** [comment]
[test katello]

**Comment by ehelms** [comment]
[test katello]

**Pull request merged**